File: app/schemas/bake_plan.py
from datetime import date
from typing import List

from pydantic import BaseModel


class BakePlanItem(BaseModel):
    product_id: int
    product_name: str
    forecast_quantity: int
    sku: str | None = None
    # Risk metrics (computed from forecast uncertainty)
    waste_risk_prob: float | None = None
    stockout_risk_prob: float | None = None
    risk_sigma: float | None = None
    interval_level_used: float | str | None = None  # 0.8, 0.95, or "fallback"
    risk_method: str | None = None  # "interval_based" | "residual_based" | "fallback"
    debug_source: str | None = None  # "prophet_interval" | "xgb_interval" | "fallback"
    sigma_clamped: bool | None = None


class BakePlanResponse(BaseModel):
    bakery_id: int
    bakery_name: str
    date: date
    items: List[BakePlanItem]

File: frontend/lib/bakePlan.ts
import { apiFetch } from "@/lib/api";

export type BakePlanItem = {
  product_id: number;
  product_name: string;
  forecast_quantity: number;
  sku?: string | null;
  // Risk metrics (computed from forecast uncertainty)
  waste_risk_prob?: number | null;
  stockout_risk_prob?: number | null;
  risk_sigma?: number | null;
  interval_level_used?: number | string | null;
  risk_method?: string | null;
  debug_source?: string | null;
  sigma_clamped?: boolean | null;
};

export type BakePlanResponse = {
  bakery_id: number;
  bakery_name: string;
  date: string;
  items: BakePlanItem[];
};

export async function fetchBakePlan(
  bakeryId: number,
  targetDate?: string
): Promise<BakePlanResponse> {
  const params = targetDate ? `?target_date=${targetDate}` : "";
  return apiFetch<BakePlanResponse>(`/api/bakeries/${bakeryId}/bake-plan${params}`);
}




